<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>742</width>
    <height>473</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>QAction，QMainWindow，QPlainTextEdit</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextUnderIcon</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPlainTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>25</y>
      <width>676</width>
      <height>326</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>class MainWindow : public QMainWindow
{
    Q_OBJECT

private:
    QLabel     *labFile;    //状态栏里使用
    QLabel     *labInfo;    //状态栏里使用
    QProgressBar    *progressBar1;  //状态栏上的进度条

    QSpinBox        *spinFontSize;  //字体大写
    QFontComboBox   *comboFontName; //字体名称

    QActionGroup  *actionGroup;     //Action分组，用于“中文”和“英文”的互斥选择

    void    buildUI(); //程序设计的UI初始化
    void    buildSignalSlots(); //手工关联信号与槽
public:
    MainWindow(QWidget *parent = 0);
    ~MainWindow();</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>742</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="actEdit_Cut"/>
    <addaction name="actEdit_Copy"/>
    <addaction name="actEdit_Paste"/>
    <addaction name="separator"/>
    <addaction name="actEdit_Undo"/>
    <addaction name="actEdit_Redo"/>
    <addaction name="separator"/>
    <addaction name="actEdit_SelectAll"/>
    <addaction name="actEdit_Clear"/>
   </widget>
   <widget class="QMenu" name="menu_F">
    <property name="title">
     <string>格式(&amp;M)</string>
    </property>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>界面语言</string>
     </property>
     <addaction name="actLang_CN"/>
     <addaction name="actLang_EN"/>
    </widget>
    <addaction name="actFont_Bold"/>
    <addaction name="actFont_Italic"/>
    <addaction name="actFont_UnderLine"/>
    <addaction name="separator"/>
    <addaction name="actSys_ToggleText"/>
    <addaction name="menu"/>
   </widget>
   <widget class="QMenu" name="menu_F_2">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="actFile_New"/>
    <addaction name="actFile_Open"/>
    <addaction name="actFile_Save"/>
    <addaction name="separator"/>
    <addaction name="actClose"/>
   </widget>
   <addaction name="menu_F_2"/>
   <addaction name="menu_E"/>
   <addaction name="menu_F"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actFile_New"/>
   <addaction name="actFile_Open"/>
   <addaction name="actFile_Save"/>
   <addaction name="separator"/>
   <addaction name="actEdit_Cut"/>
   <addaction name="actEdit_Copy"/>
   <addaction name="actEdit_Paste"/>
   <addaction name="separator"/>
   <addaction name="actEdit_Undo"/>
   <addaction name="actEdit_Redo"/>
   <addaction name="separator"/>
   <addaction name="actFont_Bold"/>
   <addaction name="actFont_Italic"/>
   <addaction name="actFont_UnderLine"/>
   <addaction name="separator"/>
   <addaction name="actLang_CN"/>
   <addaction name="actLang_EN"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
  <action name="actEdit_Cut">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/200.bmp</normaloff>:/icons/images/200.bmp</iconset>
   </property>
   <property name="text">
    <string>剪切</string>
   </property>
   <property name="toolTip">
    <string>剪切到粘贴板</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actEdit_Copy">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/202.bmp</normaloff>:/icons/images/202.bmp</iconset>
   </property>
   <property name="text">
    <string>复制</string>
   </property>
   <property name="toolTip">
    <string>复制到粘贴板</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actEdit_Paste">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/204.bmp</normaloff>:/icons/images/204.bmp</iconset>
   </property>
   <property name="text">
    <string>粘贴</string>
   </property>
   <property name="toolTip">
    <string>从粘贴板粘贴</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actFont_Bold">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/500.bmp</normaloff>:/icons/images/500.bmp</iconset>
   </property>
   <property name="text">
    <string>粗体</string>
   </property>
   <property name="toolTip">
    <string>粗体</string>
   </property>
  </action>
  <action name="actFont_Italic">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/502.bmp</normaloff>:/icons/images/502.bmp</iconset>
   </property>
   <property name="text">
    <string>斜体</string>
   </property>
   <property name="toolTip">
    <string>斜体</string>
   </property>
  </action>
  <action name="actFont_UnderLine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/504.bmp</normaloff>:/icons/images/504.bmp</iconset>
   </property>
   <property name="text">
    <string>下划线</string>
   </property>
   <property name="toolTip">
    <string>下划线</string>
   </property>
  </action>
  <action name="actClose">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/132.bmp</normaloff>:/icons/images/132.bmp</iconset>
   </property>
   <property name="text">
    <string>关闭</string>
   </property>
   <property name="toolTip">
    <string>关闭本窗口</string>
   </property>
  </action>
  <action name="actSys_ToggleText">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>显示按钮文字</string>
   </property>
   <property name="toolTip">
    <string>显示工具栏按钮文字</string>
   </property>
  </action>
  <action name="actEdit_Clear">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/212.bmp</normaloff>:/icons/images/212.bmp</iconset>
   </property>
   <property name="text">
    <string>清空</string>
   </property>
   <property name="toolTip">
    <string>清空</string>
   </property>
  </action>
  <action name="actEdit_Undo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/206.bmp</normaloff>:/icons/images/206.bmp</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="toolTip">
    <string>撤销上次编辑操作</string>
   </property>
   <property name="statusTip">
    <string>撤销上次编辑操作</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actEdit_Redo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/208.bmp</normaloff>:/icons/images/208.bmp</iconset>
   </property>
   <property name="text">
    <string>重做</string>
   </property>
   <property name="toolTip">
    <string>重做上次操作</string>
   </property>
   <property name="statusTip">
    <string>重做上次操作</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actEdit_SelectAll">
   <property name="text">
    <string>全选</string>
   </property>
   <property name="toolTip">
    <string>全选</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actFile_New">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/100.bmp</normaloff>:/icons/images/100.bmp</iconset>
   </property>
   <property name="text">
    <string>新建</string>
   </property>
   <property name="toolTip">
    <string>新建文件</string>
   </property>
   <property name="statusTip">
    <string>新建一个文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actFile_Open">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/122.bmp</normaloff>:/icons/images/122.bmp</iconset>
   </property>
   <property name="text">
    <string>打开...</string>
   </property>
   <property name="toolTip">
    <string>打开文件</string>
   </property>
   <property name="statusTip">
    <string>打开一个文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actFile_Save">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/104.bmp</normaloff>:/icons/images/104.bmp</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>保存修改</string>
   </property>
   <property name="statusTip">
    <string>没有实际保存文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actLang_CN">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/CN.jpg</normaloff>:/icons/images/CN.jpg</iconset>
   </property>
   <property name="text">
    <string>中文</string>
   </property>
   <property name="toolTip">
    <string>中文界面</string>
   </property>
   <property name="statusTip">
    <string>切换使用汉语界面</string>
   </property>
  </action>
  <action name="actLang_EN">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icons/images/timg2.jpg</normaloff>:/icons/images/timg2.jpg</iconset>
   </property>
   <property name="text">
    <string>English</string>
   </property>
   <property name="toolTip">
    <string>English Interface</string>
   </property>
   <property name="statusTip">
    <string>切换使用英语界面</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actEdit_Cut</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_Copy</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_Paste</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_Clear</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_Undo</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_Redo</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>undoAvailable(bool)</signal>
   <receiver>actEdit_Undo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>redoAvailable(bool)</signal>
   <receiver>actEdit_Redo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actEdit_SelectAll</sender>
   <signal>triggered()</signal>
   <receiver>textEdit</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textEdit</sender>
   <signal>modificationChanged(bool)</signal>
   <receiver>actFile_Save</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>330</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
